### PR TITLE
Bdd/timebase related elements

### LINKED ISSUE #187: Add test ensuring clockMode is mandatory when timeBase=clock

### PR DESCRIPTION
**Pull request by gilbrechbuhler**
The PR itself adds only a test ensuring that clockMode is mandatory when timeBase=clock. However it completes the spec-conformance document from R60 to R73b because R69 to R73a were already tested.

Fix #187

### CODE REVIEW

File: testing/bdd/features/validation/timeBase_clock_clockMode_mandatory.feature [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  # SPEC-CONFORMANCE: R73b
+  Scenario: Invalid ttp:clockMode
+    Given an xml file <xml_file>
+    When it has ttp:timeBase attribute <time_base>
+    And it has ttp:clockMode attribute <clock_mode>
+    Then document is invalid
+
+    Examples:
+    | time_base | clock_mode    |
+    | clock     |               |
```
Comment (nigelmegitt) [inline testing/bdd/features/validation/timeBase_clock_clockMode_mandatory.feature]: Is it worth also adding a test for an empty `clock-mode`, since the code is able to check for that?

File: testing/bdd/templates/timeBase_clock_clockMode_mandatory.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" ?>
+<tt:tt
+        ebuttp:sequenceIdentifier="testSequence001"
+        ebuttp:sequenceNumber="1"
+        ttp:timeBase="{{ time_base }}"
+{% if clock_mode %}
+  {% if clock_mode == "*?Empty?*" %}
+        ttp:clockMode=""
```
Comment (nigelmegitt) [inline testing/bdd/templates/timeBase_clock_clockMode_mandatory.xml]: Previous comment about empty `clockMode` related to this line. On second thoughts, should this line deliberately omit adding the `clockMode` attribute, rather than adding an empty one?

Comment (gilbrechbuhler) [inline testing/bdd/templates/timeBase_clock_clockMode_mandatory.xml]: When there is nothing in the `Examples` table, the `{% if clock_mode %}` is false and if you look at the template, then ttp:clockMode is never here. If we put `*?Empty?*` in the examples tables, then it is there and is `ttp:clockMode=""`. I can add an `*?Empty?*` test in the example table, then both cases will be covered.

Comment (gilbrechbuhler) [inline testing/bdd/templates/timeBase_clock_clockMode_mandatory.xml]: Done

Comment (nigelmegitt) [inline testing/bdd/templates/timeBase_clock_clockMode_mandatory.xml]: Thanks, apologies I always get confused by the empty/missing thing first time I read these. Not sure why!

**Comment by nigelmegitt** [comment]
Looks good to me.
